libs/sqlserver/langchain_sqlserver/vectorstores.py

```diff
@@ -225,31 +229,86 @@ def __init__(
         self._embedding_store = self._get_embedding_store(self.table_name, self.schema)
         self._create_table_if_not_exists()
 
-    def _can_connect_with_entra_id(self) -> bool:
-        """Determine if Entra ID authentication can be used.
+    def _get_connection_url(self, conn_string: str) -> str:
+        if conn_string is None or len(conn_string) == 0:
+            logging.error("Connection string value is None or empty.")
+            raise ValueError("Connection string value cannot be None.")
+
+        args = conn_string.split(";")
```

Comment:
What happens when we have an invalid conn_string like, misplaced ";", "," and "=".
Are we doing any checks before to ensure that the conn_string will not be faulty?

Reply:
Updated the code to catch other exceptions as opposed to having it catch just keyerror exception. As stated in a different comment, for actual connection string validation, sqlalchemy handles this.